### PR TITLE
nix-doc: 0.6.4 -> 0.6.5

### PR DESCRIPTION
**Pull request by lf-**
Update nix-doc to have additional API compatibility for Nix 2.20 (still builds and works on previous versions as always).

cc https://github.com/NixOS/nixpkgs/pull/285264

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/package-management/nix-doc/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   pname = "nix-doc";
-  version = "0.6.4";
+  version = "0.6.5";
 
   src = fetchFromGitHub {
     rev = "v${version}";
     owner = "lf-";
     repo = "nix-doc";
-    sha256 = "sha256-yL0oG0NiQ7OdGQ/kZxQbSbNphKapu5HBFNP5E2fVe+Y=";
+    sha256 = "sha256-9cuNzq+CBA2jz0LkZb7lh/WISIlKklfovGBAbSo1Mgk=";
```
Comment (kirillrdy) [inline pkgs/tools/package-management/nix-doc/default.nix:11]:
```suggestion
    hash = "sha256-9cuNzq+CBA2jz0LkZb7lh/WISIlKklfovGBAbSo1Mgk=";
```

File: pkgs/tools/package-management/nix-doc/default.nix [hunk 2]
```diff
@@ -29,7 +29,7 @@ rustPlatform.buildRustPackage rec {
     RUSTFLAGS = "-Z relro-level=partial";
   };
 
-  cargoSha256 = "sha256-4bzLZt45ZLTZyZPZ4Nkvz7mNe4oqOIoaZUbCbNWBKG0=";
+  cargoSha256 = "sha256-CHagzXTG9AfrFd3WmHanQ+YddMgmVxSuB8vK98A1Mlw=";
```
Comment (kirillrdy) [inline pkgs/tools/package-management/nix-doc/default.nix:32]:
```suggestion
  cargoHash = "sha256-CHagzXTG9AfrFd3WmHanQ+YddMgmVxSuB8vK98A1Mlw=";
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/package-management/nix-doc/default.nix (hunk 1 above)
Comment (lf-) [reply pkgs/tools/package-management/nix-doc/default.nix:11]: In my personal opinion this sha256->hash migration being gone about by bothering individual maintainers piecemeal, without any explicit writing down of a real deprecation or migration plan, or indeed auto migrating the entire code base, is the wrong way of going about it. Implementing tooling to either require migration in a pr check or do it automatically for you is something the nixpkgs architecture team should be doing.

Thanks for the feedback, I'm declining to take action on it absent policy.

Comment (kirillrdy) [reply pkgs/tools/package-management/nix-doc/default.nix:11]: appreciate your sincerity ! I don't mind either way ( re hash migration )

there is this in nixpkgs manual https://nixos.org/manual/nixpkgs/stable/#compiling-rust-applications-with-cargo

> . cargoHash256 is used for traditional Nix SHA-256 hashes, such as the one in the example above. cargoHash should instead be used for [SRI](https://www.w3.org/TR/SRI/) hashes

**Review by wegank — APPROVED** [review]

**Pull request merged**